### ISSUE TITLE
Lz4Raw compression failed if input is tiny

### ISSUE THREAD

**Issue by dantengsky** [jorgecarleitao/parquet2#117]
When writing parquet file with the compression option `Compression::Lz4Raw`, if the input buffer is tiny,  the writer fails with error

`Error: General("underlying IO error: Compression failed")`

e.g.

if we change the following lines
https://github.com/jorgecarleitao/parquet2/blob/f8eb2436d5b39614139fc728050a6f375781f813/tests/it/main.rs#L75-L78
to
~~~rust
        "id" => {
            let expected = vec![4];
            let expected = expected.into_iter().map(Some).collect::<Vec<_>>();
            Array::Int32(expected)
~~~

`cargo test int32_lz4` will output 

```
---- write::int32_lz4 stdout ----
Error: General("underlying IO error: Compression failed")
.....
```

**Comment by dantengsky** [comment]
Sorry,  PR opened in the incorrect repo :(

**Comment by jorgecarleitao** [comment]
Closed by #118
